cmd/rook/ceph/osd.go

```diff
@@ -300,6 +300,10 @@ func updateLocationWithNodeLabels(location *[]string, nodeLabels map[string]stri
 
 // Parse the devices, which are sent as a JSON-marshalled list of device IDs with a StorageConfig spec
 func parseDevices(devices string) ([]osddaemon.DesiredDevice, error) {
+	if devices == "" {
+		return []osddaemon.DesiredDevice{}, nil
```

Comment:
```suggestion
		return nil, nil
```
What about just `nil`? It's shorter and remains consistent with the return L310.

Reply:
There is no error in this case though. If we return nil when the code expects a slice, there could be an issue later in the code.

Reply:
Understood.

Reply:
Are there still changes you see?

Reply:
Not really but I don't see why we would have an issue later in the code since for me `[]osddaemon.DesiredDevice{}` is no different than `nil`. Nil is the empty value of the struct AFAIK. But I don't want to block this for so little.